### PR TITLE
[improve][schema] Add admin cli for testCompatibility

### PR DESCRIPTION
**Pull request by congbobo184**
### Motivation

1. add admin cli `testCompatibility`
2. add `admin.schemas().testCompatibility()` test
### Modifications

1. add admin cli `testCompatibility`
2. add `admin.schemas().testCompatibility()` test
### Verifying this change
add test for it
- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [x] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: 
https://github.com/congbobo184/pulsar/pull/14

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdSchemas.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ public CmdSchemas(Supplier<PulsarAdmin> admin) {
         jcommander.addCommand("delete", new DeleteSchema());
         jcommander.addCommand("upload", new UploadSchema());
         jcommander.addCommand("extract", new ExtractSchema());
+        jcommander.addCommand("test-compatibility", new TestCompatibility());
```
Comment (tisonkun) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdSchemas.java:45]: nit: avoid wordy

```suggestion
        jcommander.addCommand("compatibility", new TestCompatibility());
```

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by lifepuzzlefun — APPROVED** [review]

**Review by poorbarcode — APPROVED** [review]

**Review by liangyepianzhou — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

**Comment by poorbarcode** [comment]
Since this patch is a help tools API, I cherry-picked it into `branch-3.0`

**Pull request merged**